packages/fxa-settings/src/components/TotpInputGroup/index.tsx

```diff
@@ -0,0 +1,232 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import classNames from 'classnames';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { CodeArray } from '../FormVerifyTotp';
+
+export type TotpInputGroupProps = {
+  codeArray: CodeArray;
+  codeLength: 6 | 8;
+  inputRefs: React.MutableRefObject<React.RefObject<HTMLInputElement>[]>;
+  localizedInputGroupLabel: string;
+  setCodeArray: React.Dispatch<React.SetStateAction<CodeArray>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  errorMessage?: string;
+};
+
+type SingleInputProps = {
+  index: number;
+};
+
+export const TotpInputGroup = ({
+  codeArray,
+  codeLength,
+  inputRefs,
+  localizedInputGroupLabel,
+  setCodeArray,
+  setErrorMessage,
+  errorMessage = '',
+}: TotpInputGroupProps) => {
+  const focusOnNextInput = (index: number) => {
+    inputRefs.current[index + 1].current?.focus();
+  };
+
+  const focusOnPreviousInput = (index: number) => {
+    inputRefs.current[index - 1].current?.focus();
+  };
+
+  const focusOnSpecifiedInput = (index: number) => {
+    inputRefs.current[index].current?.focus();
+  };
+
+  const handleBackspace = async (index: number) => {
+    let currentCodeArray = [...codeArray];
+    currentCodeArray[index] = undefined;
+    await setCodeArray(currentCodeArray);
+    if (index > 0) {
+      focusOnPreviousInput(index);
+    } else focusOnSpecifiedInput(index);
+  };
+
+  const handleDelete = async (index: number) => {
+    let currentCodeArray = [...codeArray];
+    if (currentCodeArray[index] !== undefined) {
+      currentCodeArray[index] = undefined;
+    } else {
+      currentCodeArray[index + 1] = undefined;
+    }
+    await setCodeArray(currentCodeArray);
+    if (index < codeLength - 1) {
+      focusOnNextInput(index);
+    } else {
+      focusOnSpecifiedInput(index);
+    }
+  };
+  // TODO in FXA-7888 review for LTR/RTL
```

Comment:
@bcolsson I've seen [some comments](https://github.com/alphamu/PinEntryEditText/issues/51) that indicate that PIN code inputs should be rendered LTR for both LTR and RTL languages. Do you know if this is accurate?

Reply:
In most RTL languages that I'm aware of, numbers are typically written LTR (though it gets more confusing if they're separated into groups of numbers - like phone numbers), so the comment linked makes sense. 

In this case, yeah I think it's accurate to use LTR for RTL languages here.